core/src/main/kotlin/pages/PageBuilder.kt

```diff
@@ -54,24 +57,52 @@ class DefaultPageBuilder(
         }
     }
 
-    private fun contentForClass(c: Class) = group(c) {
-        header(1) { text(c.name) }
-        c.inherited.takeIf { it.isNotEmpty() }?.let {
-            header(2) { text("SuperInterfaces") }
-            linkTable(it)
+    private fun PageContentBuilder.headerForClasslike(c: ClassLike) =
+        header(1) {
+            text(
+                "${when (c) {
+                    is Class -> "class "
+                    is Enum -> "enum "
+                    else -> ""
+                }}${c.name}"
+            )
         }
-        c.commentsData.forEach {
+
+    private fun PageContentBuilder.contentForEnumEntries(e: Enum) =
+        block("Entries", 2, ContentKind.Properties, e.entries, e.platformData) { entry ->
+            link(entry.name, entry.dri)
+            contentForComments(entry)
+        }
+
+    private fun PageContentBuilder.contentForComments(d: Documentable) =
+        d.commentsData.forEach {
             it.children.forEach {
                 header(3) { text(it.toHeaderString()) }
                 comment(it.root)
                 text("\n")
             }
         }
+
+    private fun PageContentBuilder.contentForCtors(c: Class) =
         block("Constructors", 2, ContentKind.Functions, c.constructors, c.platformData) {
             link(it.name, it.dri)
             signature(it)
             text(it.briefDocTagString)
         }
+
+    private fun contentForClasslike(c: ClassLike) = group(c) {
+        headerForClasslike(c)
+
+        if (c is Enum) contentForEnumEntries(c)
+
+        c.inherited.takeIf { it.isNotEmpty() }?.let {
+            header(2) { text("SuperInterfaces") }
+            linkTable(it)
+        }
+        contentForComments(c)
+
+        if (c is Class) contentForCtors(c)
```

Comment:
Unneeded if we create two separate methods (see two comments above)

Reply:
I've splitted it up into separate methods.